test/dialect/postgresql/test_types.py

```diff
@@ -3673,6 +3680,490 @@ class DateTimeTZRangeRoundTripTest(_DateTimeTZRangeTests, _RangeTypeRoundTrip):
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_contains(self):
+        self._test_clause(
+            self.col.contains(self._data_str()),
+            "data_table.multirange @> %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_contained_by(self):
+        self._test_clause(
+            self.col.contained_by(self._data_str()),
+            "data_table.multirange <@ %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_overlaps(self):
+        self._test_clause(
+            self.col.overlaps(self._data_str()),
+            "data_table.multirange && %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_strictly_left_of(self):
+        self._test_clause(
+            self.col << self._data_str(),
+            "data_table.multirange << %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+        self._test_clause(
+            self.col.strictly_left_of(self._data_str()),
+            "data_table.multirange << %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_strictly_right_of(self):
+        self._test_clause(
+            self.col >> self._data_str(),
+            "data_table.multirange >> %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+        self._test_clause(
+            self.col.strictly_right_of(self._data_str()),
+            "data_table.multirange >> %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_not_extend_right_of(self):
+        self._test_clause(
+            self.col.not_extend_right_of(self._data_str()),
+            "data_table.multirange &< %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_not_extend_left_of(self):
+        self._test_clause(
+            self.col.not_extend_left_of(self._data_str()),
+            "data_table.multirange &> %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_adjacent_to(self):
+        self._test_clause(
+            self.col.adjacent_to(self._data_str()),
+            "data_table.multirange -|- %(multirange_1)s",
+            sqltypes.BOOLEANTYPE,
+        )
+
+    def test_union(self):
+        self._test_clause(
+            self.col + self.col,
+            "data_table.multirange + data_table.multirange",
+            self.col.type,
+        )
+
+    def test_intersection(self):
+        self._test_clause(
+            self.col * self.col,
+            "data_table.multirange * data_table.multirange",
+            self.col.type,
+        )
+
+    def test_different(self):
+        self._test_clause(
+            self.col - self.col,
+            "data_table.multirange - data_table.multirange",
+            self.col.type,
+        )
+
+
+class _MultiRangeTypeRoundTrip(fixtures.TablesTest):
+    __requires__ = "range_types", "psycopg_only_compatibility"
+    __backend__ = True
+
+    def extras(self):
```

Comment:
**mike bayer** (zzzeek) wrote:

what is this part for?   What happens if I dont do these things?  This looks like part of the implementation of the dialect is inside the test suite?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3728

Reply:
Ah, when originally working on this I took inspiration from the PostgreSQL _RangeTypeRoundTrip tests:
https://github.com/sqlalchemy/sqlalchemy/blob/efd4e6dd8a73b956c860d796606f8e6ad652c292/test/dialect/postgresql/test_types.py#L3493-L3511

Since only the `psycopg` dialect support a Multirange type, I could remove the extraneous `psycopg_extras` and `extras`, if you think it would be cleaner.

Reply:
so i did not realize this is just the datatype returned by the driver, so since here we're targeting to match the existing range support, we can do that for now.

Reply:
left as is.